File: LeetCode/2009. Minimum Number of Operations to Make Array Continuous/solution.cpp
class Solution {
public:
    int minOperations(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        int n = nums.size(), left = 0, right = 0, distinct = 0, res = 0;
        for (; left < n && right < n; left++){
            while (right < n && nums[right] <= nums[left]+n-1){
                if (right == 0 || nums[right] != nums[right-1])
                    distinct++;
                right++;
            }
            res = max(res, distinct);
            if (left == n-1 || nums[left] != nums[left+1])
                distinct--;
        }
        return n-res;
    }
};

File: LeetCode/2009. Minimum Number of Operations to Make Array Continuous/solution.py
class Solution:
    def minOperations(self, nums: List[int]) -> int:
        nums.sort()
        sz = len(nums)
        cnt = Counter()
        res = sz
        distinct = 0
        right = 0
        for left in range(sz):
            while right < sz and nums[right] < nums[left] + sz:
                if cnt[nums[right]] == 0:
                    distinct += 1
                cnt[nums[right]] += 1
                right += 1
            res = min(res, sz-distinct)
            cnt[nums[left]] -= 1
            if cnt[nums[left]] == 0:
                distinct -= 1
        return res